search/search-client/src/main/java/io/camunda/search/clients/CamundaSearchClients.java

```diff
@@ -313,23 +431,35 @@ protected <T> SearchQueryResult<T> ensureSingeResultIfNecessary(
     return result;
   }
 
-  protected <T, Q extends TypedSearchQuery<?, ?>> SearchQueryResult<T> doSearch(
-      final SearchEntityReader<T, Q> reader, final Q query) {
+  protected <T, Q extends TypedSearchQuery<?, ?>> Optional<T> doGet(
+      final Function<ResourceAccessChecks, T> applier) {
     try {
-      return doReadWithResourceAccessController(a -> reader.search(query, a));
-    } catch (final ResourceAccessDeniedException denied) {
-      final var missingAuthorization = denied.getMissingAuthorization();
-      LOG.trace(
-          "Unauthorized to perform a Search Query with permission '{}' on resource '{}', returning an empty search query result",
-          missingAuthorization.permissionType(),
-          missingAuthorization.resourceType(),
-          denied);
-      return SearchQueryResult.empty();
+      return Optional.ofNullable(doGetWithResourceAccessController(applier));
+    } catch (final TenantAccessDeniedException e) {
+      LOG.trace("Forbidden to access tenant, returning null", e);
+      return Optional.empty();
     }
   }
 
   protected <T> T doReadWithResourceAccessController(
       final Function<ResourceAccessChecks, T> applier) {
     return resourceAccessController.doSearch(securityContext, applier);
   }
+
+  protected <T> T doGetWithResourceAccessController(
+      final Function<ResourceAccessChecks, T> applier) {
+    return resourceAccessController.doGet(securityContext, applier);
+  }
+
+  protected CamundaSearchException entityByKeyNotFoundException(
+      final String entityType, final long key) {
+    return new CamundaSearchException(
+        ERROR_ENTITY_BY_KEY_NOT_FOUND.formatted(entityType, key), Reason.NOT_FOUND);
+  }
+
+  protected CamundaSearchException entityByIdNotFoundException(
+      final String entityType, final String id) {
+    return new CamundaSearchException(
+        ErrorMessages.ERROR_ENTITY_BY_ID_NOT_FOUND.formatted(entityType, id), Reason.NOT_FOUND);
```

Comment:
🔧  we statically import one of the ErrorMessages but not the other - let's be consistent here